controllers/useraccount/useraccount_controller.go

```diff
@@ -484,6 +500,93 @@ func (r *Reconciler) deleteUser(ctx context.Context, userAcc *toolchainv1alpha1.
 	return true, nil
 }
 
+// Returns `true` if the associated resources (configMap, role and role-binding) created for a user by console is deleted, `false` otherwise with the underlying error.
+// This function only looks for these resources in the namespace - openshift-console-user-settings -
+func (r *Reconciler) deleteUserResources(ctx context.Context, userID string) (bool, error) {
+
+	// new user will have a label introduced which can help retrieve resources, otherwise older ones will have to rely on the name of the resource being of type `user-settings-<UID>`
+	// delete ConfigMap, Role and RoleBinding
+	if deleted, err := r.deleteConfigMap(ctx, userID); err != nil || deleted {
+		return deleted, err
+	}
+	if deleted, err := r.deleteRole(ctx, userID); err != nil || deleted {
+		return deleted, err
+	}
+	if deleted, err := r.deleteRoleBinding(ctx, userID); err != nil || deleted {
+		return deleted, err
+	}
+	return false, nil
+}
+
+func (r *Reconciler) deleteConfigMap(ctx context.Context, uid string) (bool, error) {
+	name := ConsoleUserSettingsPrefix + uid
+	logger := log.FromContext(ctx)
+	logger.Info(fmt.Sprintf("deleting configmap with name %s", name))
+	toDelete := &corev1.ConfigMap{}
+	toDelete.SetGroupVersionKind(schema.GroupVersionKind{
+		Kind: "ConfigMap",
+	})
+	configMap, err := getObjectByName(ctx, r.Client, name, toDelete)
+	if err != nil || configMap == nil {
```

Comment:
I don't think we should hide the error here. We should ignore 404 errors (see above) but otherwise return the error here. And only if we didn't find the object to delete and there is no error (again, we ignore 404) we should try to use the label.